chore(observatory): update the FAQ for the new API V2 (#11954)
mdn/yari: copy/observatory/faq.md

## Has the HTTP Observatory API been updated to use the new tests?

> ~~Not yet. The API will continue using the old test infrastructure for a while,

Reviewer: I'm not really sure why you'd want to keep this text but in a strikethrough form.

Author: We don't, I left it in for a reason that escaped me ;)

therefore you will see some small differences between test scores returned by
the API and the website. The API will be updated to use the new tests in a
near-future iteration.~~

> Yes. It lives under a new URL (see below) and we have cleaned up the API

Reviewer: ```suggestion
Yes. The new v2 API is available at an updated URL — `https://observatory-api.mdn.mozilla.net/api/v2/scan` — and the
```

> response a little bit. A POST call to

Reviewer: ```suggestion
response is now more concise. For example, a [`POST`](/en-US/docs/Web/HTTP/Methods/POST) request to
```

`https://observatory-api.mdn.mozilla.net/api/v2/scan?host=mdn.net` will return a
JSON payload like this:

```json
{
  "id": 53494870,
  "details_url": "https://developer.mozilla.org/en-US/observatory/analyze?host=mdn.dev",
  "algorithm_version": 4,
  "scanned_at": "2024-10-11T13:21:36.453Z",
  "error": null,
  "grade": "A+",
  "score": 105,
  "status_code": 200,
  "tests_failed": 0,
  "tests_passed": 10,
  "tests_quantity": 10
}
```

> We have removed a number of fields that were not too useful for CI integration,

Reviewer: ```suggestion
We have removed several fields that were of limited use for CI integration,
```

> like the complete listing of headers. The important metrics like `score` and the

Reviewer: ```suggestion
like the complete response header listing. The important metrics like `score` and
```

> `grade` are still there and we encourage you to migrate to the new API as soon

Reviewer: ```suggestion
`grade` are still included. We encourage you to migrate to the new API as soon
```

> as possible. The old one will be shut down Oct 31, 2024.

Reviewer: ```suggestion
as possible: The v1 endpoint will be shut down on October 31, 2024.
```

> For reference, the v1 API returned

Reviewer: ```suggestion
For reference, the v1 API returned a JSON payload with the following structure:
```

```json
{
  "algorithm_version": 3,
  "end_time": "Fri, 11 Oct 2024 13:19:31 GMT",
  "grade": "A+",
  "hidden": false,
  "likelihood_indicator": "LOW",
  "response_headers": {
    "Accept-Ranges": "none",

>     ...

Reviewer: Because you can't put comments in JSON, this is not valid syntax and might even fall foul of the linter. Maybe replace it with a placeholder like

```
"Another-Header": "another-value"
```

Or something?

I'm not 100% sure about this. Maybe it isn't important.

Author: agreed
